hs_core/views/__init__.py

```
@@ -344,18 +387,53 @@ class AddGroupForm(forms.Form):
+        except:
+            # unable to connect to the database
+            raise DatabaseError("unable to connect to the database HSAccess.")
+
+        try:
+            if t == 'add_view_user':
+                frm = AddUserForm(data=request.POST)
+                if frm.is_valid():
+                    res.view_users.add(frm.cleaned_data['user'])
+                    new_user = frm.cleaned_data['user']
+                    new_uuid = ha_obj.get_user_uuid_from_login(str(new_user.username))
+                    ha_obj.share_resource_with_user(res.short_id, new_uuid, 'ro')
+            elif t == 'add_edit_user':
+                frm = AddUserForm(data=request.POST)
+                if frm.is_valid():
+                    res.edit_users.add(frm.cleaned_data['user'])
+                    new_user = frm.cleaned_data['user']
+                    new_uuid = ha_obj.get_user_uuid_from_login(str(new_user.username))
+                    ha_obj.share_resource_with_user(res.short_id, new_uuid, 'rw')
+            elif t == 'add_owner':
+                frm = AddUserForm(data=request.POST)
+                if frm.is_valid():
+                    res.owners.add(frm.cleaned_data['user'])
+                    new_user = frm.cleaned_data['user']
+                    new_uuid = ha_obj.get_user_uuid_from_login(str(new_user.username))
+                    ha_obj.share_resource_with_user(res.short_id, new_uuid, 'own')
+            elif t == 'make_public':
+                #if res.metadata.has_all_required_elements():
+                if res.can_be_public:
+                    res.public = True
+                    ha_obj.make_resource_public(res.short_id)
+                    res.save()
+            elif t == 'make_private':
+                res.public = False
+                ha_obj.make_resource_not_public(res.short_id)
+                res.save()
+        except:
```

another bare except and again a feeling that failing here is just not right...